Kinder/Registration.xaml.cs

```diff
@@ -28,119 +28,155 @@ public RegistrationWindow()
                 ErrorNameSurname.Visibility = Visibility.Visible;
                 Name.BorderBrush = Brushes.Red;
                 Surname.BorderBrush = Brushes.Red;
-            } else if (RegValidation.CheckIfFieldIsValid(Name.Text) || RegValidation.CheckIfFieldIsValid(Surname.Text))
+            }
+            catch (WhiteSpaceDetectedException ex)
             {
-                ErrorNameSurname.Text = "Name or Surname is not valid!";
+                ErrorNameSurname.Text = ex.Message;
                 ErrorNameSurname.Visibility = Visibility.Visible;
                 Name.BorderBrush = Brushes.Red;
                 Surname.BorderBrush = Brushes.Red;
-            } else
+            }
+
+            try                                                     ///Email try/catch block
             {
-                ErrorNameSurname.Visibility = Visibility.Hidden;
-                Name.BorderBrush = Brushes.Green;
-                Surname.BorderBrush = Brushes.Green;
+                RegValidation.CheckTextBoxInput(Email.Text);
+                RegValidation.CheckIfEmailValid(Email.Text);
+                User.CheckIfUserAlreadyExists(Email.Text);
+                ErrorEmail.Visibility = Visibility.Hidden;
+                Email.BorderBrush = Brushes.Green;
                 count++;
             }
-
-            if (RegValidation.CheckTextBoxInput(Email.Text))            ///Email
+            catch (EmptyFieldException ex)
             {
-                ErrorEmail.Text = "Enter Email!";
+                ErrorEmail.Text = ex.Message;
                 ErrorEmail.Visibility = Visibility.Visible;
                 Email.BorderBrush = Brushes.Red;
-            } else if(User.CheckIfUserAlreadyExists(Email.Text))
+            }
+            catch (InvalidEmailException ex)
             {
-                ErrorEmail.Text = "User already exists!\nTry to log in!";
+                ErrorEmail.Text = ex.Message;
                 ErrorEmail.Visibility = Visibility.Visible;
                 Email.BorderBrush = Brushes.Red;
-            } else if (!RegValidation.CheckIfEmailValid(Email.Text))
+            }
+            catch (UserAlreadyExistsException ex)
             {
-                ErrorEmail.Text = "Email is not valid!";
+                ErrorEmail.Text = ex.Message;
                 ErrorEmail.Visibility = Visibility.Visible;
                 Email.BorderBrush = Brushes.Red;
-            } else
+            }
+
+            try                                                     ///Phonenumber try/catch block
             {
-                ErrorEmail.Visibility = Visibility.Hidden;
-                Email.BorderBrush = Brushes.Green;
+                RegValidation.CheckTextBoxInput(PhoneNumber.Text);
+                RegValidation.CheckIfPhoneNumberValid(PhoneNumber.Text);
+                User.CheckIfUserAlreadyExists(PhoneNumber.Text);
+                ErrorPhoneNumber.Visibility = Visibility.Hidden;
```

Comment:
isn't visibility hidden by default?

Reply:
if it isn't maybe set up visibility before try block, since setting up visibility isn't "trying" to detect error

Reply:
that would go for other similar sequences in other try blocks too

Reply:
Hm, you may be right. I moved visibility (the green one) before try block